experimental/go/gsdk.go

```diff
@@ -0,0 +1,96 @@
+package gsdk
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	ig = newInternalGsdk()
+)
+
+// Start kicks off communication threads, heartbeats, etc.
+func Start() {
+	ig.startInternal()
+}
+
+// StartStartWithDebugLogs is like Start(), but with debug logs enabled
+func StartWithDebugLogs() {
+	ig.debugLogs = true
+	ig.startInternal()
+}
+
+// ReadyForPlayers transitions the state to standingBy which tells the PlayFab service that the game server is ready to accept players
+// Be aware that it is a *blocking* call, it unblocks when the game server is allocated (transitions to Active)
+func ReadyForPlayers() bool {
+	ig.startInternal()
+	if ig.currentGameState != GameStateActive {
+		ig.currentGameState = GameStateStandingBy
+		<-ig.transitionToActiveEvent
+	}
+	return ig.currentGameState == GameStateActive
+}
+
+// MarkAllocated is an experimental method that transitions the state to Active
+func MarkAllocated() error {
+	if ig.currentGameState != GameStateStandingBy {
+		return errors.New("game server is not in the standing by state to be marked allocated")
+	}
+	ig.currentGameState = GameStateActive
+	ig.transitionToActiveEvent <- struct{}{}
+	return nil
+}
+
+// LogMessage adds a custom log message to the GSDK log output
+func LogMessage(s string) {
+	ig.startInternal()
+	logInfo(s)
+}
+
+// GetGameServerConnectionInfo gets information (ipAddress and ports) for connecting to the game server, as well as the ports the game server should listen on.
+func GetGameServerConnectionInfo() GameServerConnectionInfo {
+	ig.startInternal()
+	return ig.configuration.GameServerConnectionInfo
+}
+
+// RegisterMaintananceCallback gets called if the server is getting a scheduled maintenance, it will get the UTC Datetime of the maintenance event as an argument.
+func RegisterMaintananceCallback(callback func(time.Time)) {
+	ig.startInternal()
+	ig.maintenanceCallback = callback
+}
+
+// RegisterHealthCallback gets called when the agent needs to check on the game's health
+func RegisterHealthCallback(callback func() bool) {
```

Comment:
So in the current setup, this functioncall is mandatory otherwise the server healthcheck will return false. Maybe we should consider the callback delegate here to be a parameter of the Start() method?

Reply:
so, on MPS side, we don't do anything if the healthcheck returns false (we should fix that). We should do something on thundernetes as well, opened this https://github.com/PlayFab/thundernetes/issues/56

The function signature is similar to the one in the other GSDKs as well, e.g. -> https://github.com/PlayFab/gsdk/blob/master/csharp/GSDK_CSharp_Standard/GameserverSDK.cs#L127

Reply:
Ok, if its consistent with the other implementations, then we shouldn't make this one special. For thundernetes we have options of actions to take, but I'll put thoughts in that issue.